### PR TITLE
VEN-1208: expire old offers cronjob. Refactor common cronjob code int…

### PR DESCRIPTION
**Pull request by haavikko**
…o base class.

### Closes :no_good_woman:
**[VEN-1208](https://helsinkisolutionoffice.atlassian.net/browse/VEN-1208):**

### CODE REVIEW

File: payments/management/command_util.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import logging
```
Comment (a-rmz) [inline payments/management/command_util.py:1]: Should this file be something like`base_expiration_command.py`?

File: payments/management/commands/expire_too_old_offers.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from django.conf import settings
+from django.db.transaction import atomic
+
+from payments.management import command_util
+from payments.models import BerthSwitchOffer
```
Comment (a-rmz) [inline payments/management/commands/expire_too_old_offers.py:5]:
```suggestion
from . import command_util
from ..models import BerthSwitchOffer
```

File: payments/management/command_util.py (hunk 1 above)
Comment (haavikko) [reply payments/management/command_util.py:1]: fixed

File: payments/management/commands/expire_too_old_offers.py (hunk 2 above)
Comment (haavikko) [reply payments/management/commands/expire_too_old_offers.py:5]: fixed

Comment (haavikko) [reply payments/management/commands/expire_too_old_offers.py]: Removed draft status from this PR

File: payments/models.py [hunk 3]
```diff
@@ -1120,6 +1143,19 @@ def clean(self):
...
+
+            # if due_date change is allowed
+            self._check_due_date(old_instance)
+
+    def _check_due_date(self, old_instance) -> None:
+        if (
+            old_instance.due_date != self.due_date
+            and old_instance.status == OfferStatus.EXPIRED
+        ):
+            raise ValidationError(_("Cannot change due date of this offer"))
```
Comment (a-rmz) [inline payments/models.py:1157]: Neat 👌

File: payments/tests/factories.py [hunk 4]
```diff
@@ -194,8 +194,8 @@ class AbstractOfferFactory(factory.django.DjangoModelFactory):
 class BerthSwitchOfferFactory(AbstractOfferFactory):
     lease = factory.SubFactory(
         BerthLeaseFactory,
-        customer=factory.SelfAttribute("..customer"),
         status=LeaseStatus.PAID,
+        customer=factory.SelfAttribute("..customer"),
```
Comment (a-rmz) [inline payments/tests/factories.py:198]: Does this make any difference?

**Comment by a-rmz** [comment]
We're not enabling it yet, right?

File: payments/tests/factories.py (hunk 4 above)
Comment (haavikko) [reply payments/tests/factories.py:198]: Fixed. That's right, the ordering doesn't matter. I think I had made the change when I started the branch and then afterwards the same change was introduced to the main branch from another commit.

**Comment by haavikko** [comment]
> We're not enabling it yet, right?

I noticed the configs in .git/workflows were missing, added them. It will be enabled in review and staging, not production.

**Review by a-rmz — APPROVED** [review]

**Pull request merged**